shell auto-configuration

joshlong opened the pull request:
this contributes `@EnableDataFlowShell` and required configuration.

markfisher commented on spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/EnableDataFlowShell.java:
```diff
@@ -0,0 +1,20 @@
+package org.springframework.cloud.dataflow.shell;
```
missing copyright header

markfisher commented on spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BootstrapConfiguration.java:
```diff
@@ -0,0 +1,120 @@
+package org.springframework.cloud.dataflow.shell.autoconfigure;
```
missing copyright header

markfisher commented:
The `spring-cloud-dataflow` build fails on this branch. Both of these test classes within the shell sub-project are failing: `StreamCommandTests` and `TaskCommandTests`

joshlong commented:
thanks @markfisher for taking a look! 

i just added the two copyright headers but im confused what you mean the build was failing. Travis was green and my local build was green (mvn -f  spring-cloud-dataflow/pom.xml  clean install ) works..

jvalkeal commented:
Tests ran fine when I tried.

markfisher commented on spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/EnableDataFlowShell.java:
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015 the original author or authors.
```
it's 2016

markfisher commented on spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java:
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 the original author or authors.
```
2016

markfisher commented on spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java:
```diff
@@ -0,0 +1,80 @@
+package org.springframework.cloud.dataflow.shell.autoconfigure;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
should be an empty line before `org.springframework.*` imports

markfisher commented on spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BootstrapConfiguration.java:
```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2015 the original author or authors.
```
2016

markfisher commented:
`spring-shell-plugin.xml` was moved under `src/test`... is it even needed anymore?

joshlong commented:
@markfisher  its' needed because the two tests that use the shell still use the old style `Bootstrap` class since creating a shell and a server in the same VM and not having auto configuration step on each other was doubly painful.

joshlong commented:
tests pass on my machine, updated headers, added space between `o.s.*` imports, and even remembered to add `@EnableDataFlowShell` to the `ShellApplication` so that `java -jar target/spring-cloud-dataflow-shell-1.0.0.BUILD-SNAPSHOT.jar` works as expected :D HURRAY

markfisher commented:
@ericbottard can you please take a quick look at this PR? If you're okay with it, you could merge it. Otherwise assuming it looks good to you, I will merge first thing in the morning.

ericbottard commented:
Will do

ericbottard commented on spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java:
```diff
@@ -109,19 +110,23 @@ public static void startUp() throws InterruptedException, IOException {
+					"--spring.jmx.enabled=false",
+					"--security.basic.enabled=false",
 					"--spring.main.show_banner=false", "--spring.cloud.config.enabled=false",
 					"--deployer.local.out-of-process=false");
```
This should be removed I believe

ericbottard commented:
Looks good to me overall

ericbottard commented:
Squashed and merged as 95c5fb1, including the removal of the property I mentioned above.

markfisher commented:
Thanks @ericbottard ! and thanks @joshlong for contributing this.... nice improvement!
